packages/compiler/src/core/compiler-code-fixes/triple-quote-indent.codefix.ts

```diff
@@ -0,0 +1,124 @@
+import { CharCode, isWhiteSpaceSingleLine } from "../charcode.js";
+import { defineCodeFix, getSourceLocation } from "../diagnostics.js";
+import type { CodeFixEdit, DiagnosticTarget } from "../types.js";
+
+export function createTripleQuoteIndentCodeFix(diagnosticTarget: DiagnosticTarget) {
+  return defineCodeFix({
+    id: "triple-quote-indent",
+    label: "Format triple-quote-indent",
+    fix: (context) => {
+      const result: CodeFixEdit[] = [];
+      const location = getSourceLocation(diagnosticTarget);
+      const startPos = location.pos + 3;
+      const endPos = location.end - 3;
+      const splitStr = "\r\n";
+
+      const lines = splitLines(location.file.text, startPos, endPos);
+
+      const firstLine = lines[0];
+      if (lines.length > 0 && firstLine.lineText.trim() === "") {
+        lines.shift();
+      }
+
+      const lastLine = lines[lines.length - 1];
+      if (lines.length > 0 && lastLine.lineText.trim() === "") {
+        lines.pop();
+      }
+
+      const minIndentLine: { startPos: number; indent: number } = lines.reduce((prev, curr) => {
+        return prev.indent < curr.indent ? prev : curr;
+      });
+      if (minIndentLine.indent <= lastLine.indent) {
+        let indentDiff = lastLine.indent - minIndentLine.indent;
+        const prefix = " ".repeat(indentDiff);
+        // start triple-quote is not on a new line
+        if (firstLine.lineText.trim() !== "") {
+          result.push(context.prependText({ ...location, pos: startPos }, splitStr + prefix));
```

Comment:
instead of adding fix for each line, can't we just have one fix for the whole string?